test/rnn_vanilla_common.hpp

```diff
@@ -540,8 +540,11 @@ void RNNFwdTrainCPUVerify(miopen::Handle& handle,
     {
         for(int h = 0; h < out_h; h++)
         {
-            assert(!std::isnan(rsvspace.at(prelayer_shift + bs * hy_stride + h)));
-            assert(!std::isinf(rsvspace.at(prelayer_shift + bs * hy_stride + h)));
+            assert(
+                !std::isnan(static_cast<double>(rsvspace.at(prelayer_shift + bs * hy_stride + h))));
+            assert(
+                !std::isinf(static_cast<double>(rsvspace.at(prelayer_shift + bs * hy_stride + h))));
```

Comment:
@apwojcik Why this is necessary? Is it for `half`, `bfloat16` or for integer type?